### DISCUSSION TITLE
How to scale my FastAPI application which I am converting from a Django application?

### DISCUSSION THREAD

**Discussion by raghavsikaria** [fastapi/fastapi#4396] [Questions] [answered]
I have a Django application that has been scaled like this: the application is hosted on 5 different servers and each instance is served on an Apache server with 24 threads.

What would a similar scale-up be for the FastAPI application? Would it be to serve it using gunicorn with 24 uvicorn workers, each deployed on those 5 servers?

also, what are other alternatives?

**Comment by jgould22** [comment, marked as answer]
The correct way to scale is the same way you would scale most web applications (outside of cloud deployments, k8s, etc).

Place a reverse proxy (you could use apache, or NGINX or HAProxy) then as you said run Gunicorn with its recommended starting settings of  `(2 x $num_cores) + 1` and adjust accordingly based on your load.
